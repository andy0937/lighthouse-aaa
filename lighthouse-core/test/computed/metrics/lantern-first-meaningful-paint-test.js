/**
 * @license Copyright 2017 Google Inc. All Rights Reserved.
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */
'use strict';

const assert = require('assert');

const trace = require('../../fixtures/traces/progressive-app-m60.json');
const devtoolsLog = require('../../fixtures/traces/progressive-app-m60.devtools.log.json');
const LanternFirstMeaningfulPaint =
  require('../../../computed/metrics/lantern-first-meaningful-paint.js');

/* eslint-env jest */
describe('Metrics: Lantern FMP', () => {
  it('should compute predicted value', async () => {
    const computedCache = new Map();
    const result = await LanternFirstMeaningfulPaint.request({trace, devtoolsLog,
      settings: {}}, {computedCache});

    expect({
      timing: Math.round(result.timing),
      optimistic: Math.round(result.optimisticEstimate.timeInMs),
      pessimistic: Math.round(result.pessimisticEstimate.timeInMs),
    }).toMatchSnapshot();
    assert.equal(result.optimisticEstimate.nodeTimings.size, 5);
    assert.equal(result.pessimisticEstimate.nodeTimings.size, 8);
    assert.ok(result.optimisticGraph, 'should have created optimistic graph');
    assert.ok(result.pessimisticGraph, 'should have created pessimistic graph');
  });
});
