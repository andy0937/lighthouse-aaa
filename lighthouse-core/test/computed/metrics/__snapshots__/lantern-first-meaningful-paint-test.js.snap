// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics: Lantern FMP should compute predicted value 1`] = `
Object {
  "optimistic": 1439,
  "pessimistic": 1583,
  "timing": 1511,
}
`;
